EVM-verifier/contracts/AliceRingToken.sol

```diff
@@ -0,0 +1,176 @@
+        MINTED // the proof has already been minted
+    }
+
+    error AlreadyMinted(string proofId);
+    error InvalidSignature();
+    error InvalidTokenAmounts();
+    error OnlyOwnerCanBurn(uint256 tokenId);
+
+    RingSignatureVerifier public verifier; // Instance of the ring signature verifier contract
+    mapping(string => Status) public mintStatus; // signatureHash => Status (computed off-chain)
+
+    constructor(address _verifier) ERC721("AliceRingToken", "ART") Ownable() {
+        verifier = RingSignatureVerifier(_verifier);
+    }
+
+    // The following functions are overrides required by Solidity.
+    function tokenURI(uint256 tokenId) public view override(ERC721, ERC721URIStorage) returns (string memory) {
+        return super.tokenURI(tokenId);
+    }
+
+    function supportsInterface(bytes4 interfaceId) public view override(ERC721, ERC721URIStorage) returns (bool) {
+        return super.supportsInterface(interfaceId);
+    }
+
+    /**
+     * @notice safeTransferFrom is disabled because the nft is a sbt
+     */
+    function safeTransferFrom(address from, address to, uint256 tokenId) public pure override(ERC721, IERC721) {
+        revert("SBT: SafeTransfer not allowed");
+    }
+
+    /**
+     * @notice transferFrom is disabled because the nft is a sbt
+     */
+    function transferFrom(address from, address to, uint256 tokenId) public pure override(ERC721, IERC721) {
+        revert("SBT: Transfer not allowed");
+    }
+
+    /**
+     * @notice approve is disabled because the nft is a sbt and cannot be transferred
+     */
+    function approve(address to, uint256 tokenId) public pure override(ERC721, IERC721) {
+        revert("SBT: Approve not allowed");
+    }
+
+    /**
+     * @notice getApproved is disabled because the nft is a sbt and cannot be transferred
+     */
+    function getApproved(uint256 tokenId) public pure override(ERC721, IERC721) returns (address operator) {
+        revert("SBT: getApproved not allowed");
+    }
+
+    /**
+     * @notice setApprovalForAll is disabled because the nft is a sbt and cannot be transferred
+     */
+    function setApprovalForAll(address operator, bool _approved) public pure override(ERC721, IERC721) {
+        revert("SBT: setApprovalForAll not allowed");
+    }
+
+    /**
+     * @notice isApprovedForAll is disabled because the nft is a sbt and cannot be transferred -> the output will always be false
+     */
+    function isApprovedForAll(address owner, address operator) public pure override(ERC721, IERC721) returns (bool) {
+        return false;
+    }
+
+    /**
+     * @notice Verifies the token amounts for each address in the ring
+     *
+     * @param token - the erc20 address of the token we are proving the ownership of
+     * @param signature - the signature object
+     */
+    function verifyTokenAmounts(address token, uint256 minBalance, Signature memory signature)
+        internal
+        returns (bool)

```

Comment:
- `@param minBalance` is missing
- this function can be restricted to view

Reply:
@Elli610 done! Silly that I missed this, thanks for pointing out...